vald/evm/evm.go

```diff
@@ -486,6 +466,42 @@ func (mgr Mgr) GetTxReceiptIfFinalized(chain nexus.ChainName, txID common.Hash,
 	return txReceipt, nil
 }
 
+// GetTxReceiptsIfFinalized retrieves receipts for provided transaction IDs, only if they're finalized.
+func (mgr Mgr) GetTxReceiptsIfFinalized(chain nexus.ChainName, txIDs []common.Hash, confHeight uint64) ([]rs.Result[*geth.Receipt], error) {
+	client, ok := mgr.rpcs[strings.ToLower(chain.String())]
+	if !ok {
+		return nil, fmt.Errorf("rpc client not found for chain %s", chain.String())
+	}
+
+	results, err := client.TransactionReceipts(context.Background(), txIDs)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(errors.With(err, "chain", chain.String()), "cannot get transaction receipts")
+	}
+
+	isFinalized := func(receipt *geth.Receipt) rs.Result[*geth.Receipt] {
+		isFinalized, err := mgr.isTxReceiptFinalized(chain, receipt, confHeight)
+
+		if err != nil {
+			if err == ethereum.NotFound {
```

Comment:
where can `mgr.isTxReceiptFinalized` return `ethereum.NotFound`?

Reply:
for got to remove that 😥